### ISSUE TITLE
/usr/share/temboard-agent/auto_configure.sh: line 54: used_a[*]: unbound variable

### ISSUE THREAD

**Issue by wsescu** [dalibo/temboard-agent#507]
Dear temBoard team

when I try to deploy a new Agent (Version temboard-agent-7.4-1.el7.noarch), I do get the following error:

```
$ sudo /usr/share/temboard-agent/auto_configure.sh https://host.example.ch:8888
Using hostname host.example.ch.
Configuring for PostgreSQL user postgres.
Configuring for cluster on port 55031.
Configuring for cluster at /app/lib/postgres/pgdata/12/P01A.
Configuring temboard-agent in /etc/temboard-agent/12/pg55031/temboard-agent.conf .
Saving auto-configuration in /etc/temboard-agent/12/pg55031/temboard-agent.conf.d/auto.conf
Failure. See /var/log/temboard-agent-auto-configure.log for details.
```

In the log I see the following error:

```
+++ ss -ln4t '( sport >= 2345 and sport <= 3000 )'
+++ grep -Po ':\K\d+'
/usr/share/temboard-agent/auto_configure.sh: line 54: used_a[*]: unbound variable
+ port=
+ catchall
+ '[' 1 -gt 0 ']'
```

Cheers
William

**Comment by pgiraud** [comment]
I could confirm that the script fails on CentOS7.

**Comment by wsescu** [comment]
Dear @pgiraud 

This might not be the best solution, but a workaround, which worked for me, is to replace the upper port of 3000 with 300000.

```
From:
mapfile -t used_a < <(ss -ln4t '( sport >= 2345 and sport <= 3000 )' | grep -Po ':\K\d+')

To:
mapfile -t used_a < <(ss -ln4t '( sport >= 2345 and sport <= 300000 )' | grep -Po ':\K\d+')
```

Cheers
William

**Comment by pgiraud** [comment]
That's a bit weird, but thanks for the tip anyway.

**Comment by wsescu** [comment]
On my system, I have no open port between 2345 and port 3000 and so the variable is empty. However, if I increase the upper port number to 300000 it will find some ports and the variable will be filled. Maybe a check is missing. e.g. if $user_a is empty, then use 2345.

```
$ mapfile -t used_a < <(ss -ln4t '( sport >= 2345 and sport <= 3000 )' | grep -Po ':\K\d+')
$ echo $used_a
              <= empty
$ mapfile -t used_a < <(ss -ln4t '( sport >= 2345 and sport <= 300000 )' | grep -Po ':\K\d+')
$ echo $used_a
3128
$ used="${used_a[*]}"
$ echo $used
3128 23980 20782 17808 20048
```

**Comment by pgiraud** [comment]
Oh, thanks, that will help us a lot.

**Comment by pgiraud** [comment]
@wsescu It would be very helpful if you could give the following patch a try.
https://github.com/dalibo/temboard-agent/pull/508/files

**Comment by wsescu** [comment]
Hello @pgiraud 

sure, I have tested it on RHEL 7 and RHEL 8 and it works perfectly now.  It picked now port 2345.

```
find_next_free_port() {
        local port
        local used_a
        local used
        mapfile -t used_a < <(ss -ln4t '( sport >= 2345 and sport <= 3000 )' | grep -Po ':\K\d+')
        # To mock ss output, use seq:
        # mapfile -t used_a < <(seq 2345 3000)
        # used="${used_a[*]}"
        used="${used_a[*]:-}"
        for port in {2345..3000} ; do
                if [[ " $used " =~ \ $port\  ]] ; then continue ; fi
                echo $port;
                return
        done
        log "No free TCP port found between 2345 and 3000. Force with env TEMBOARD_PORT."
        return 1
}
```


```
$ sudo /usr/share/temboard-agent/auto_configure.sh https://h50ly20.suvanet.ch:8888
Using hostname host.example.ch.
Configuring for PostgreSQL user postgres.
Configuring for cluster on port 55031.
Configuring for cluster at /app/lib/postgres/pgdata/12/P02A.
Configuring temboard-agent in /etc/temboard-agent/12/pg55031/temboard-agent.conf .
Saving auto-configuration in /etc/temboard-agent/12/pg55031/temboard-agent.conf.d/auto.conf
Configuring temboard-agent to run on port 2345.
Enabling systemd unit temboard-agent@12-pg55031.service.

Success. You can now start temboard-agent using:

    systemctl start temboard-agent@12-pg55031.service

For registration, use secret key xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx .
See documentation for detailed instructions.
```

Cheers
William

**Comment by pgiraud** [comment]
Fixed via #508
